eventespresso/barista
# Add New Feature Flags Admin UI

For the E2E tests that @alexkuc is developing, we needed to remove the feature flags filter that was in the "Barista for Event Espresso" add-on, that essentially turned ALL of the flags on (for development purposes), because he needs manual control over them which he will do via WP-CLI. 

This of course meant that there was no longer a way to turn those filters on for anyone that doesn't use WP-CLI. 

This PR:

- adds a new tab to the EE Maintenance admin page named "Feature Flags"
- adds a series of switches for enabling or disabling individual feature flags
- refactors some of the logic in the EDTR (specifically the bulk edit feature) so that things can be modified directly via the new feature flags admin.
- removed the show/hide button for the bulk edit functionality
- the bulk edit UI will now always show when using the Entity List table view in the EDTR
- fixes some other style issues I noticed


Here's what the new admin page looks like:
![image](https://github.com/eventespresso/barista/assets/1751030/c7493f44-5933-4663-984e-16a6c22d1865)

And here's what the Event Dates List and filters look like in table view after the changes:
![image](https://github.com/eventespresso/barista/assets/1751030/f227dd23-a8e6-4659-b931-2be20890ee27)

### testing notes

- log into the admin and go to the WP Plugins admin page
- activate the "Barista for Event Espresso" plugin
- go to EE Maintenance admin page and click on the "Features Flag" tab
- toggle ALL switches off and click update
- [x] confirm that changes update correctly
- try toggling one or more feature flag switches ON and click update
- [x] confirm that changes update correctly
- [x] confirm that navigating away and returning to page still shows same settings
- turn ALL feature flags off again and click update
- in another browser tab, go the the Events admin page and add/edit an event
- toggle one of the feature flags to ON and click update
- refresh the event editor and look for the new feature / functionality
- [ ] confirm that toggling the feature flag switch affects the UI accordingly
- continue toggling each feature flag switch in the new UI and checking that the corresponding feature/functionality has changed in the EDTR
- [ ] confirm that everything works as expected
